### DISCUSSION TITLE
Minimal project setup for frozen lib files using VS code and PyMakr

### DISCUSSION THREAD

**Discussion by stenjo** [micropython/micropython#13547] [Using MicroPython]
I've spent quite a good amount of time trying to get my head around using frozen code on my Micropython project running on a Wemos D1 mini (V.4) - basically an EPS8266 Generic port. I have found some good guides and examples worth reading ([Micropython manifest files](https://docs.micropython.org/en/latest/reference/manifest.html) and [micropython-example-boards](https://github.com/micropython/micropython-example-boards)) but still cannot seem to get something simple working in my setup.

My  project structure looks like this:
```
root
|-lib/
| |-micropython
| |-myOwnModule 
|-boot.py
|-main.py
|-myOwnFIle.py

```
In addition to this, I need a 
- `manifest.py` file to tell `make` what part of my code should be frozen and what not,
- `Makefile` to help me run the compilation of firmware.
and probably a couple more files too that I do not know about.

The guides I have read so far does not hit the mark on setting this up real simple. Maybe there is not a simple way? 
My questions are:
- What is the minimum set of files needed to make the lib folder frozen?
- Where would the firmware be located, and could it be downloaded to target using PyMakr `Upload to device` - command? (Ctr-Shift-S/Cmd-Shift-S)?
- How should the Makefile look?

**Comment by jimmo** [comment]
@stenjo 

> Makefile to help me run the compilation of firmware.

You do not need a Makefile (although the `micropython-example-boards` provide one as a convenience). What you need instead is a `mpconfigboard.mk` that can be loaded from the main build, such that you can do

```
cd ports/stm32
make BOARD_DIR=/path/to/my/board
```

(change stm32 to whatever port you're using).

The main purpose of `mpconfigboard.mk` in your case will be to set `FROZEN_MANIFEST` to your `manifest.py` (likely in the same directory).

This file needs to include the default one, and then use `module()` and/or `package()` to add your additional frozen code.

e.g. `package("myOwnModule", base_path="lib")`
